### PR TITLE
fix(grep): references to cypress-grep

### LINKED ISSUE #24696: @cypress/grep readme references previous cypress-grep package
### Subject

readme

### Description

The documentation at https://github.com/cypress-io/cypress/tree/develop/npm/grep has mostly been copied from the previous `cypress-grep` repo. Apart from the initial installation instructions a lot of references to the older package still exist so are confusing. For example:

- Support File installation instructions are wrong, commands should be `import cypressGrep from '@cypress/grep'`
  - Also worth noting here is that types are incorrect in this case as no declaration is provided for this export
- Similar issue with the Config File section
- Typescript support section again mentions the wrong package `cypress-grep` instead of `@cypress/grep`
- Debugging section mentions wrong package name when searching for the package
- The example repos use the older package which is also confusing, perhaps they can be marked with a warning about that until updating to work with the newer package

Hopefully these are all simple changes to make to avoid any confusion during the migration like we had.

### LINKED ISSUE #24967: @cypress/grep internal read.me is correct - NPM references deprecated cypress-grep package
### Current behavior

When searching for cypress grep online you're correctly pointed to [@cypress/grep npm package](https://www.npmjs.com/package/@cypress/grep), but the readme is referencing the deprecated cypress-grep plugin that is no longer maintained specifically how to import in the support file & setup in the config file.  

### Desired behavior

The current npm package should be updated to reflect what is in the Cypress @cypress/grep readme. 

### Test code to reproduce

none

### Cypress Version

11.2.0

### Node version

16

### Operating System

windows 11

### Debug Logs

```shell
none
```

### Other

none

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes #24967
- Closes #24696

### Additional details
fix references still pointing to `cypress-grep`

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by jordanpowell88** [comment]
Reopening a simplified version of [this PR](https://github.com/cypress-io/cypress/pull/24992#issuecomment-1461019607) without tests (as the tests were blocking the original PR)

File: npm/grep/README.md [hunk 1]
```diff
@@ -595,11 +595,11 @@ The above scenario was confusing - did you want to find all tests with title con
...
-Version >= 3 of cypress-grep _only_ supports Cypress >= 10.
+Version >= 3 of @cypress/grep _only_ supports Cypress >= 10.
 
 ## Small Print
 
 License: MIT - do anything with the code, but don't blame me if it does not work.
 
 Support: if you find any problems with this module, email / tweet /
-[open issue](https://github.com/cypress-io/cypress/issues) on Github
+[open issue](https://github.com/cypress-io/cypress/issues) on Github
```
Comment (emilyrohrbough) [inline npm/grep/README.md]:
```suggestion
[open issue](https://github.com/cypress-io/cypress/issues) on Github.
```

File: npm/grep/src/support.js [hunk 2]
```diff
@@ -228,7 +228,7 @@ if (!Cypress.grep) {
    *  // remove all current grep settings
    *  // and run all tests
    *  Cypress.grep()
-   * @see "Grep from DevTools console" https://github.com/cypress-io/cypress-grep#devtools-console
```
Comment (emilyrohrbough) [inline npm/grep/src/support.js:231]: [system-tests/projects/nx-monorepo](https://github.com/cypress-io/cypress/pull/26108/files#diff-b09164f9746e441615bb535496c1fdcbb9e4a7621866049d8a4ec3fa863f6a6e) was this file added on accident?

Comment (jordanpowell88) [reply npm/grep/src/support.js:231]: yes

Comment (jordanpowell88) [reply npm/grep/src/support.js:231]: Fixed

**Review by emilyrohrbough — APPROVED** [review]

**Review by mschile — APPROVED** [review]

**Pull request merged**